pkg/planner/core/casetest/cascades/memo_test.go

```diff
@@ -63,12 +65,11 @@ func TestDeriveStats(t *testing.T) {
 		tk.Session().GetSessionVars().PlanColumnID.Store(0)
 		builder, _ := plannercore.NewPlanBuilder().Init(tk.Session().GetPlanCtx(), ret.InfoSchema, hint.NewQBHintHandler(nil))
 		p, err := builder.Build(ctx, nodeW)
+		p.SCtx().GetSessionVars().StmtCtx.OriginalSQL = tt
 		require.NoError(t, err, tt)
-		p, err = plannercore.LogicalOptimizeTest(ctx, builder.GetOptFlag(), p.(base.LogicalPlan))
+		p, err = plannercore.LogicalOptimizeTest(ctx, builder.GetOptFlag()|rule.FlagCollectPredicateColumnsPoint, p.(base.LogicalPlan))
```

Comment:
since it‘s in test case and the builder seems doesn't have a flag injection interface.
![image](https://github.com/user-attachments/assets/d43c5e72-5aa6-40c7-9ea0-6f529ab1a6d4)
while for user common portal, we always adjust the syncload flag in doOptimize here like below
![image](https://github.com/user-attachments/assets/28220b42-1de5-48ca-80c4-0a7c91e5815b)